contracts/rewards/StableUsnMaker.sol

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+// P1 - P3: OK
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../interfaces/ISwap.sol";
+
+contract StableUsnMaker is Ownable {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    ISwap public stableSwap;
+    address public lpMaker;
+    address private immutable usn;
+    address private immutable usdc;
+    address private immutable usdt;
+
+    event LogSetLPMaker(address oldLPMaker, address newLPMaker);
+    event LogSetStableSwap(address oldStableSwap, address newStableSwap);
+
+    event LogWithdrawFees();
+
+    event LogSwapStableTokenToUsn(address stableTokenToConvertToUsn, uint256 stableTokenAmount);
+
+    event LogUsnSentToLPMaker(uint256 usnAmount);
+
+    constructor(
+        address _stableSwap,
+        address _lpMaker,
+        address _usn,
+        address _usdt,
+        address _usdc 
+    ) public {
+        stableSwap = ISwap(_stableSwap);
+        lpMaker = _lpMaker;
+        usn = _usn;
+        usdt = _usdt;
+        usdc = _usdc;
+    }
+
+    function setStableSwap(ISwap _stableSwap) public onlyOwner {
```

Comment:
lets remove this:
if we want to change stableSwap contract, better to deploy a new StableUsnMaker 
this is to reduce the attack surface.